test/servinge2e/helpers.go

```diff
@@ -44,3 +53,128 @@ func withServiceReadyOrFail(ctx *test.Context, service *servingv1.Service) *serv
+	if err != nil {
+		return nil, err
+	}
+	httpClient := http.Client{}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	stat, err := extracMetrictData(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return stat, nil
+}
+
+func extracMetrictData(body io.Reader) (*telemetryStat, error) {
+	var parser expfmt.TextParser
+	metricFamilies, err := parser.TextToMetricFamilies(body)
+	if err != nil {
+		return nil, fmt.Errorf("reading text format failed: %v", err)
+	}
+	tel := telemetryStat{}
+	pm := prometheusMetric(metricFamilies, "serverless_telemetry")
+	if pm == nil {
+		return nil, fmt.Errorf("could not get metric family from prometheus exported metrics")
+	}
+	services := getMetricValueByTypeLabel("service", pm)
+	if services == nil {
+		return nil, errors.New("could not get metric `service` from prometheus exported metrics")
+	}
+	tel.services = *services
+	configs := getMetricValueByTypeLabel("configuration", pm)
+	if configs == nil {
+		return nil, errors.New("could not get metric `configuration` from prometheus exported metrics")
+	}
+	tel.configurations = *configs
+	revisions := getMetricValueByTypeLabel("revision", pm)
+	if revisions == nil {
+		return nil, errors.New("could not get metric `revision` from prometheus exported metrics")
+	}
+	tel.revisions = *revisions
+	routes := getMetricValueByTypeLabel("route", pm)
+	if routes == nil {
+		return nil, errors.New("could not get metric `route` from prometheus exported metrics")
+	}
+	tel.routes = *routes
+	return &tel, nil
+}
+
+func prometheusMetric(metricFamilies map[string]*ioprometheusclient.MetricFamily, key string) []*ioprometheusclient.Metric {
+	if metric, ok := metricFamilies[key]; ok && len(metric.Metric) > 0 {
+		return metric.Metric
+	}
+	return nil
+}
+
+func getMetricValueByTypeLabel(label string, metrics []*ioprometheusclient.Metric) *float64 {
+	for _, metric := range metrics {
+		if len(metric.Label) == 0 {
+			return nil
+		}
+		// we expect one label
+		if metric.Label[0] == nil {
+			return nil
+		}
+		if metric.Label[0].Name == nil || metric.Label[0].Value == nil {
+			return nil
+		}
+		if (*metric.Label[0].Name) == "type" && (*metric.Label[0].Value) == label {
+			return metric.Gauge.Value
+		}
+	}
+	return nil
+}
+
+func setupMetricsRoute(caCtx *test.Context) error {
+	metricsRoute := &v1.Route{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      "metrics",
+			Namespace: "openshift-serverless",
+		},
+		Spec: v1.RouteSpec{
+			Host: "knative-openshift-metrics",
+			Port: &v1.RoutePort{
+				TargetPort: intstr.FromString("8383"),
+			},
+			Path: "/metrics",
+			To: v1.RouteTargetReference{
+				Kind: "Service",
+				Name: "knative-openshift-metrics",
+			},
+			TLS: &v1.TLSConfig{
+				Termination:                   v1.TLSTerminationEdge,
+				InsecureEdgeTerminationPolicy: v1.InsecureEdgeTerminationPolicyAllow,
+			},
+			WildcardPolicy: v1.WildcardPolicyNone,
```

Comment:
Are these required? I'd think no.

Reply:
I guess they are meant to allow non-secure connections within the cluster and so taht tls is terminated to the edge, I guess not relevant if we are hitting with http only.